core/domain/question_services.py

```diff
@@ -756,14 +759,28 @@ def update_question(
             question.
         commit_message: str. A description of changes made to the
             question.
+        version: int. Version of the question coming from payload while
+            updating question.
 
     Raises:
         ValueError. No commit message was provided.
+        Exception. Version of the incoming question and current question
+            differs.
     """
     if not commit_message:
         raise ValueError(
             'Expected a commit message, received none.')
     updated_question = apply_change_list(question_id, change_list)
+    if (version is not None) and (version < updated_question.version):
```

Comment:
Just to confirm, if the current version of question is 4 and it got updated to 5. Won't these values be 4 and 5, but the condition shouldn't throw an error? or is the 'version' value in payload the incremented version?

Reply:
Actually the version update happens when commit is performed, here the `updated_question.version` still have the version of the current question  and `version` have the version coming from the payload. `apply_change_list` functions changes other values of the question but it's version is updated when a commit is performed

Reply:
Sorry for the delay.
Ah ok, thanks for confirming, lgtm!